src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -160,10 +160,16 @@ export class CellModel extends Disposable implements ICellModel {
 			if (!this._attachments) {
 				this._attachments = {};
 			}
-			// TO DO: Check if name already exists and message the user?
-			this._attachments[name] = attachment;
-			this.sendChangeToNotebook(NotebookChangeType.CellMetadataUpdated);
+			// Check if name already exists and get a unique name
+			if (this._attachments[name] && this._attachments[name][mimeType] !== attachment[mimeType]) {
+				name = this.getUniqueImageName(name.substring(0, name.indexOf('.')), name.substring(name.indexOf('.') + 1));
```

Comment:
Using 
```
const extName = path.extname(name);
name = this.getUniqueImageName(path.basename(name, extName), extName);
```

would be cleaner/safer (for example right now you're splitting on the first . which isn't correct, the extension is from the last .)